packages/cozy-konnector-libs/libs/linkBankOperations.js

```diff
@@ -174,40 +174,49 @@ class Linker {
         }
       })
     })
-    .then(() => {
-      const unlinkedBills = this.getUnlinkedBills(result)
-      const billsGroups = this.groupBills(unlinkedBills)
-
-      const combinations = flatten(
-        Object
-          .values(billsGroups)
-          .map(billsGroup => this.generateBillsCombinations(billsGroup))
-      )
+    .then(async () => {
+      let found
 
-      const combinedBills = combinations.map(
-        combination => this.combineBills(...combination)
-      )
+      do {
+        found = false
+
+        const unlinkedBills = this.getUnlinkedBills(result)
+        const billsGroups = this.groupBills(unlinkedBills)
+
+        const combinations = flatten(billsGroups.map(
+          billsGroup => this.generateBillsCombinations(billsGroup)
+        ))
+
+        const combinedBills = combinations.map(
+          combination => this.combineBills(...combination)
+        )
 
-      const promises = combinedBills.map(combinedBill => {
-        return findDebitOperation(this.cozyClient, combinedBill, options, allOperations)
-          .then(debitOperation => {
-            if (!debitOperation) return
+        for (const combinedBill of combinedBills) {
+          const debitOperation = await findDebitOperation(
+            this.cozyClient,
+            combinedBill,
+            options,
+            allOperations
+          )
 
+          if (debitOperation) {
+            found = true
             log('debug', combinedBill, 'Matching bills combination')
             log('debug', debitOperation, 'Matching debit debitOperation')
 
-            combinedBill.originalBills.forEach(originalBill => {
+            combinedBill.originalBills.forEach(async originalBill => {
               const res = result[originalBill._id]
               res.debitOperation = debitOperation
 
-              this.addBillToOperation(originalBill, debitOperation)
+              await this.addBillToOperation(originalBill, debitOperation)
```

Comment:
No need for await here, updateAttributes only does synchronous operations and returned Promise.resolve not to change from the previous code. We can remove the Promise.resolve from updateAttributes and remove the async/await here.